### PR TITLE
 Show an error message if a support user already exists

### PR DESCRIPTION
**Pull request by tijmenb**
##  Context

Currently, when you enter a duplicate email address in the provider user form on support we raise a 500 error.

## Changes proposed in this pull request

This changes it to a proper validation error.

## Link to Trello card

https://trello.com/c/4dsmwvAW/1519-show-a-sensible-error-if-a-user-tries-to-add-the-same-provider-email-twice

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/models/support_interface/provider_user_form.rb [hunk 1]
```diff
@@ -34,5 +35,15 @@ def self.from_provider_user(provider_user)
         provider_ids: provider_user.provider_ids,
       )
     end
+
+  private
+
+    def email_is_unique
```
Comment (stevehook) [inline app/models/support_interface/provider_user_form.rb]: Do we need to downcase `email_address` here (or earlier) to ensure that we can't duplicates that differ only by case?

Comment (tijmenb) [reply app/models/support_interface/provider_user_form.rb]: Great point - fixed in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1124/commits/416a5772dbb765558cdbf45432ef7a37b7cb2e54.

File: spec/system/support_interface/managing_provider_users_spec.rb [hunk 2]
```diff
@@ -11,7 +11,11 @@
     and_i_click_the_users_link
     and_i_click_the_manange_provider_users_link
     and_i_click_the_add_user_link
-    and_i_enter_the_users_email_and_dsi_uid
+    and_i_enter_an_existing_email
+    and_i_click_add_user
+    then_i_see_an_error
+
+    and_i_enter_the_users_email
```
Comment (tvararu) [inline spec/system/support_interface/managing_provider_users_spec.rb]:
```suggestion
    when_i_enter_the_users_email
```

Minor

File: app/models/support_interface/provider_user_form.rb [hunk 3]
```diff
@@ -19,6 +21,10 @@ def save
       )
     end
 
+    def email_address=(raw_email_address)
+      @email_address = raw_email_address.downcase
+    end
```
Comment (stevehook) [inline app/models/support_interface/provider_user_form.rb]: nice

**Pull request merged**